linux/ibus-keyman/meson.build

```diff
@@ -0,0 +1,44 @@
+project('ibus-keyman', 'c', 'cpp',
+        version: run_command('cat', '../../VERSION.md', check: true).stdout().strip(),
+        license: 'GPL-2+',
+        # default_options : ['buildtype=release',
+        #                    'cpp_std=c++14',
+        #                    'b_vscrt=static_from_buildtype',
+        #                    'warning_level=2'],
+        meson_version: '>=0.53.0')
+cc = meson.get_compiler('c')
+conf = configuration_data()
+
+ibus =        dependency('ibus-1.0',             version: '>= 1.2.0')
+gtk =         dependency('gtk+-3.0',             version: '>= 2.4')
+x11 =         dependency('x11',                  version: '>= 1.6')
+json_glib =   dependency('json-glib-1.0',        version: '>= 1.0')
+
+kmnkbp_lib = cc.find_library(
+  'libkmnkbp0',
+  dirs: [
+    '@0@/../../core/build/arch/debug/src'.format(meson.current_source_dir()),
+    '@0@/../../core/build/arch/release/src'.format(meson.current_source_dir())
+  ]
+)
+
+env = find_program('env')
+
+# Check if we have patched ibus (https://github.com/ibus/ibus/pull/2440)
+if cc.has_header_symbol('ibus.h', 'IBUS_CAP_PREFILTER', dependencies: [ibus], required: false)
+  conf.set('IBUS_HAS_PREFILTER', 1)
+endif
+
+conf.set('HAVE_CONFIG_H', 1)
+configure_file(output : 'config.h',
+               configuration : conf)
+
+core_include_dirs = [
+  include_directories('../../core/src'),
+  include_directories('../../core/src/kmx'),
```

Comment:
```suggestion
  include_directories(core_dir / 'src'),
  include_directories(core_dir / 'src/kmx'),
```

Reply:
Unfortunately this doesn't work - meson complains about absolute paths:

```
ERROR: Tried to form an absolute path to a source dir. You should not do that but use relative paths instead.
```

Reply:
Great work meson. We can I guess make a "core_source_dir" which is all dotty but it'll have to be defined for each meson.build file. Sometimes it is a little too opinionated for its own good!

Reply:
We'll still get the benefit elsewhere of using `core_dir`, just not for `include_directories` I think?